thenewboston_node/blockchain/views/blockchain_states_meta.py

```diff
@@ -0,0 +1,68 @@
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework.exceptions import NotFound, ParseError
+from rest_framework.response import Response
+from rest_framework.viewsets import ViewSet
+
+from thenewboston_node.blockchain import exceptions
+from thenewboston_node.blockchain.pagination import CustomLimitOffsetPagination
+from thenewboston_node.blockchain.serializers.blockchain_states_meta import BlockchainStatesMetaSerializer
+from thenewboston_node.business_logic.blockchain.base import BlockchainBase
+from thenewboston_node.business_logic.blockchain.file_blockchain import FileBlockchain
+
+SORT_DIRECTION_QUERY_MAP = {
+    'asc': 1,
+    'desc': -1,
+}
+
+
+class BlockchainStatesMetaViewSet(ViewSet):
+    serializer_class = BlockchainStatesMetaSerializer
+    pagination_class = CustomLimitOffsetPagination
+    lookup_value_regex = r'(-)?\d+'
+
+    @extend_schema(parameters=[OpenApiParameter('id', location=OpenApiParameter.PATH, description='Block number')])
+    def retrieve(self, request, pk: str = None):
+        assert pk is not None
+        block_number = int(pk)
+        blockchain = self._get_blockchain()
+
+        try:
+            blockchain_state_meta = blockchain.get_blockchain_state_meta_by_block_number(block_number)
+        except ValueError:
+            raise NotFound()
+
+        return Response(self.serializer_class(blockchain_state_meta).data)
+
+    @extend_schema(
+        parameters=[
+            OpenApiParameter('ordering', location=OpenApiParameter.QUERY, description='asc/desc', default='asc'),
+        ]
+    )
+    def list(self, request):  # noqa: A003
```

Comment:
please, consider using generic views some how (maybe we need to override some methods) to avoid reimplementation a lot of stuff